Jenkins_jobs/WorkspaceInfo.groovy

```diff
@@ -0,0 +1,157 @@
+                        /* This is commented out because it takes way too long to return 
+                         * it isn't a top priority
+                         * dir /s /-c
+                         */
+                        case 'windows':
+                        //     clones[machineName] = {
+                        //         node(machineName) {
+                        //                 cleanWs()
+                        //                 NodeHelper nodeHelper = new NodeHelper();
+                        //                 workspaceDirectory += "\\workspace";
+                        //                 println workspaceDirectory
+                        //                 // TODO: get path to directory from the api
+                        //                 String workspaceStats = sh (
+                        //                         script: '#!/bin/sh -e\n'  'du -sh '  workspaceDirectory,
+                        //                         returnStdout: true).trim()
+
+                        //                 String script = 'set x; du -sh '  workspaceDirectory  '\\* | sort -rn';
+                        //                 String subdirectories = sh (
+                        //                         script: 'script',
+                        //                         returnStdout: true).trim()
+                                 
+                                 
+                        //                 println beautify(machineName,workspaceDirectory, workspaceStats,subdirectories);                                 
+                                 
+                        //             cleanWs()
+                        //             }
+                        //         }
+                        //     }
+                        //     break;
+                        case 'zos':
+                        default:
+                            println ("Support for ${kernelName} is yet to be implemented");
+                            break;
+                    }
+                }
+            }
+        }
+        currentInstance = null;
+        computers = null;
+
+        cleanWs()
+    }
+}
+
+@NonCPS
+def beautify(machineName, workspaceDirectory, workspaceStats, subdirectories) {
+    workspaceStats = workspaceStats.replaceAll("\\s+", "        ");
+    subdirectories = subdirectories.replace(workspaceDirectory + "/", "");
+    String output = "\n\n=======================================================================\n";
+    output += "Disk stats for ${machineName}"
+    output += "\nWorkspace:\n${workspaceStats}";
+    def subdirectoriesArray = subdirectories.split("\n");
+    for (String line : subdirectoriesArray) {
+        if (!line.contains("tmp")) {
+            line = line.replaceAll("\\s+", "    ")
+            output += "\n    ${line}";
+        }
+    }
+    output += "\n=======================================================================\n\n";
+}
+
+parallel clones
```

Comment:
Please add a newline at the end of the file

Reply:
Done!